src/navigation/OnboardingStack.tsx

```diff
@@ -1,62 +1,154 @@
+import { TouchableOpacity, StyleSheet, View } from "react-native"
 import {
   TransitionPresets,
   createStackNavigator,
+  StackNavigationOptions,
 } from "@react-navigation/stack"
+import { useNavigation } from "@react-navigation/native"
+import { useTranslation } from "react-i18next"
+import { SvgXml } from "react-native-svg"
+
+import { OnboardingScreen, OnboardingScreens } from "./index"
+import { GlobalText } from "../components"
 
 import NotificationPermissions from "../Onboarding/NotificationPermissions"
-import EnableExposureNotifications from "../Onboarding/EnableExposureNotifications"
+import ActivateProximityTracing from "../Onboarding/ActivateProximityTracing"
 import Welcome from "../Onboarding/Welcome"
 import PersonalPrivacy from "../Onboarding/PersonalPrivacy"
 import EulaModal from "../Onboarding/EulaModal"
-import NotificationDetails from "../Onboarding/NotificationDetails"
-import ShareDiagnosis from "../Onboarding/ShareDiagnosis"
-import { OnboardingScreen, OnboardingScreens } from "./index"
 import LanguageSelection from "../More/LanguageSelection"
+import Introduction from "../Onboarding/Introduction"
+import PhoneRemembersDevices from "../Onboarding/PhoneRemembersDevices"
+import GetNotified from "../Onboarding/GetNotified"
+import ValueProposition from "../Onboarding/ValueProposition"
 
-const SCREEN_OPTIONS = {
-  headerShown: false,
-}
+import { Icons } from "../assets"
+import { Spacing, Colors, Typography } from "../styles"
 
 type OnboardingStackParams = {
   [key in OnboardingScreen]: undefined
 }
 
 const Stack = createStackNavigator<OnboardingStackParams>()
 
-const OnboardingStack: FunctionComponent = () => (
-  <Stack.Navigator screenOptions={SCREEN_OPTIONS}>
-    <Stack.Screen name={OnboardingScreens.Welcome} component={Welcome} />
-    <Stack.Screen
-      name={OnboardingScreens.EulaModal}
-      component={EulaModal}
-      options={{ ...TransitionPresets.ModalTransition }}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.PersonalPrivacy}
-      component={PersonalPrivacy}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.NotificationDetails}
-      component={NotificationDetails}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.ShareDiagnosis}
-      component={ShareDiagnosis}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.NotificationPermissions}
-      component={NotificationPermissions}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.EnableExposureNotifications}
-      component={EnableExposureNotifications}
-    />
-    <Stack.Screen
-      name={OnboardingScreens.LanguageSelection}
-      component={LanguageSelection}
-    />
-  </Stack.Navigator>
-)
+const OnboardingStack: FunctionComponent = () => {
+  const { t } = useTranslation()
+  const navigation = useNavigation()
+
+  const onboardingScreenOptions: StackNavigationOptions = {
+    headerShown: false,
+  }
+
+  const HeaderRight = (stepNumber: number) => {
```

Comment:
How do we feel about creating an enum or a union type to hold the screens? It feels like a stretch to have indexes since we are hardcoding the values in it. Could be something like:
``` typescript
enum OnboardingStep = {
  ValuePropositionStep = 1
  ...
}
type OnboardingStep = "ValuePropositionStep" | ...
```
This way you could tie a number with a meaning and we're not passing only cold numbers around, this is probably not a critical improvement but it will give you some safety here that we won't miss any screen if one gets added. Happy to talk more about it. You could probably even tie it to a subset of the screen names that will  be on the navigator, in case an update happens on the type system only and then we will get a warning here that we have an incomplete resolver.

Reply:
Great call thoughts on ac0fedbc50f023157dca92e24e8cb4b556a7fa3f?

Reply:
Actually @aledustet I think this implements what you suggested in case you're interested.

Reply:
Sorted it out thanks!